### ISSUE TITLE
secret_environment_variables as list of key-value pairs

### ISSUE THREAD

**Issue by EwertonES** [google-github-actions/deploy-cloud-functions#287]
### TL;DR

On README, it says that the optional parameter secret_environment_variables is a list of key-value pairs, but the example shows only a single key-value pair. What should I do if I have more than one secret that I want to associate with my Function?

I tried creating a list like:
```
secret_environment_variables:
  - 'PG_PASSWORD=projects/xxxxxxxxxx/secrets/PG_PASSWORD'
  - 'PG_HOST=projects/xxxxxxxxxx/secrets/PG_HOST'
```

But it gives an error: 
`The workflow is not valid. .github/workflows/main.yml (Line: 42, Col: 13): A sequence was not expected`

### Expected behavior

When a list of key-value pairs is passed in the YAML file, it should associate all the secrets present in the list to the Cloud Function.

### Observed behavior

I didn't find a way to pass secrets as list.

### Action YAML

```yaml
name: CI

on:
  push:
    branches: [ main ]
  pull_request:
    branches: [ main ]
  workflow_dispatch:

jobs:
  build:
    runs-on: ubuntu-latest

    permissions:
      contents: 'read'
      id-token: 'write'

    steps:
      - uses: 'actions/checkout@v2'

      - name: 'Authenticate in Google Cloud'
        uses: 'google-github-actions/auth@v0'
        with:
          workload_identity_provider: ''
          service_account: ''
          project_id: ''

      - name: 'Set up Cloud SDK'
        uses: 'google-github-actions/setup-gcloud@v0'

      - id: 'deploy'
        name: 'Deploy Cloud Function'
        uses: 'google-github-actions/deploy-cloud-functions@v0'
        with:
          name: ''
          runtime: 'python39'
          entry_point: 'main'
          memory_mb: '2048'
          description: ''
          vpc_connector: ''
          secret_environment_variables:
            - ''
            - ''
            - ''
            - ''
            - ''
            - ''
```

### Log output

```text
The workflow is not valid. .github/workflows/main.yml (Line: 42, Col: 13): A sequence was not expected
```

### Additional information

First time writing an issue, sorry if I missed something.

**Comment by sethvargo** [comment]
Hi @EwertonES 

This is a limitation of GitHub Actions. The `with` block only supports `key: value` pairs, so you cannot have nested structures like that. However, you can create a multi-line string to make it easier to read:

```yaml
secret_environment_variables: |-
  PG_PASSWORD=projects/xxxxxxxxxx/secrets/PG_PASSWORD
  PG_HOST=projects/xxxxxxxxxx/secrets/PG_HOST
```

Let me know if you have additional questions, and sorry there's not much we can do about this in our action directly. You can also file feedback with GitHub by filing a support ticket.

**Comment by mattdsteele** [comment]
Just a heads up - based on my testing, you'll need to comma-delimit your variables or you'll run into errors. So something like this should work:

```yaml
secret_environment_variables: |-
  PG_PASSWORD=projects/xxxxxxxxxx/secrets/PG_PASSWORD,
  PG_HOST=projects/xxxxxxxxxx/secrets/PG_HOST
```

**Comment by sethvargo** [comment]
Hi @mattdsteele you shouldn't need to have a comma at the end. If you're experiencing other behavior, please open an issue and complete the issue template so we can debug.
